src/libraries/System.Private.CoreLib/src/System/IO/BufferedStream.cs

```diff
@@ -959,7 +960,8 @@ public override void Write(ReadOnlySpan<byte> buffer)
             {
                 // We do not expect buffer sizes big enough for an overflow, but if it happens, lets fail early:
                 totalUserbytes = _writePos + buffer.Length;
-                useBuffer = (totalUserbytes + buffer.Length < (_bufferSize + _bufferSize));
+                // Allow current totalUserbytes up to int.MaxValue by using uint arithmetic operation for totalUserbytes + buffer.Length
+                useBuffer = ((uint)totalUserbytes + buffer.Length < (_bufferSize + _bufferSize));
```

Comment:
Is there a test you can write for this?

Reply:
Hi @stephentoub , yes, actually I just removed one verification used test case from [commit](https://github.com/dotnet/runtime/pull/53338/commits/1d89790e6cdfa19296eed3742430bca44e4579ff) because I think it needs to allocate a large byte array object (although it is temp variable). So can we accept to add these kind of test case ?

Reply:
> So can we accept to add these kind of test case ?

Yes, e.g. https://github.com/dotnet/runtime/blob/72ff0df4638ced1cc2b15c92c055f842c51e32d2/src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonWriterTests.cs#L2922.  Note that it's outerloop, is conditioned to only run on 64-bit, and early exits the test if it's unsuccessful in allocating the required amount.

Reply:
@stephentoub Tests added, please review.